webpack.config.js

```diff
@@ -0,0 +1,72 @@
+const path = require("path");
+const HTMLWebpackPlugin = require("html-webpack-plugin");
+const { CleanWebpackPlugin } = require("clean-webpack-plugin");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+
+const isDev = process.env.NODE_ENV === "development";
+const isProd = !isDev;
+
+const filename = (ext) =>
+  isDev ? `[name].${ext}` : `[name].[contenthash].${ext}`;
+
+module.exports = {
+  context: path.resolve(__dirname, "src"),
+  mode: "development",
+  entry: "./js/index.js",
+  output: {
+    filename: filename("js"),
+    path: path.resolve(__dirname, "dist"),
+    clean: true,
+  },
+  devServer: {
+    historyApiFallback: true,
+    open: true,
+    hot: true,
+    static: {
+      directory: path.resolve(__dirname, "dist"),
+    },
+    compress: true,
+    port: 3000,
+  },
+  plugins: [
+    new HTMLWebpackPlugin({
+      filename: "index.html",
+      template: path.resolve(__dirname, "src/index.html"),
+      minify: {
+        collapseWhitespace: isProd,
+      },
+    }),
+    new CleanWebpackPlugin(),
+    new MiniCssExtractPlugin({
+      filename: filename("css"),
+    }),
+  ],
+  module: {
+    rules: [
+      {
+        test: /\.html$/i,
+        loader: "html-loader",
+      },
+      {
+        test: /\.css$/i,
+        use: [
+          {
+            loader: MiniCssExtractPlugin.loader,
+            options: {
+              hmr: isDev,
+            },
+          },
+          "css-loader",
+        ],
+      },
```

Comment:
Do you have `.css` files?

Reply:
sass-loader Loads a Sass/SCSS file and compiles it to CSS.
After that they are processed by the another loaders and plugins.

Reply:
I can't see sass-loader here